2020-01-Winter/4_trees/README.md

```diff
@@ -116,21 +145,61 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
-
-Source: TODO :bug:
+### 2. SOLUTION 2
+
+Source:
+
+#### Recursion Solution
+```python3
+class Solution:
+  def findNode(self,node, count, result):
+      if result[0] != -1:
```

Comment:
_**Strongly**_ recommend avoiding using real values as placeholders; it's entirely possible that our BST contains `-1` and that it is the value we are looking for, which would make our solution more likely to break. 

In a situation like array indexing, `-1` is a great placeholder because `-1` is not a valid array index in most languages. Here, we must be careful. 

Instead of using `-1`, we leave the list as empty, and then _append_ our desired value.
```suggestion
      # have we found the kth node and added its value?
      if result:
```